design.md

```diff
@@ -0,0 +1,94 @@
+# y2c
+
+## Description
+
+Automatically import events from Github repos into Google Calendar.
+
+## Overview
+
+Github Action that reads event data from specific YAML files (`y2c.yml`) in a GitHub repo and calls the Google Calendar API to create/update events. The action is only triggered when master is pushed and the files containing event data change. Event attendees can either be individual's email addresses or a Google Group email address.
+
+
+## Not in scope
+
+* Managing members of the Google Groups. If members need to be added to the Google Group, this will need to happen via the Google Groups UI.
+* Creating/Updating events on groups.io
+* Using Groups from groups.io.
+
+## Format of event files (`y2c.yml`)
+
+```yaml
+id: 'event-id'
+organizer: 'dan@linuxfoundation.com' # email address of the organizer
+calendarId: 'calendar-id'           # Calendar Identifier
+title: 'Event title'
+description: 'Event Description'
+location: 'zoom-url'                # Are we using the Zoom URL as location?
+start:
+  date: '2020-02-10'                # Event start date
+  time: '9:00am'                    # Event start time
+  timeZone: 'America/New_York'
+end:
+  date: '2020-02-10'                # Event end date
+  time: '11:00am'                   # Event end time
+  timeZone: 'America/New_York'
+attendees:
+  - email: joe@example.com          # Guest email
+    name: 'Joe User'                # Guest name (optional)
+  - email: jane@example.com
+    name: 'Jane User'
+reminders:
+  - method: 'email'                 # Send email reminder 10 minutes before event
+    minutes: 10
+  - method: 'popup'                 # Show popup reminder 1 hour before event
+    minutes: 60
+```
+
+#### Notes
+* The user will have to specify an event ID, so changes to the `y2c.yml` file trigger an update on Google Calendar, instead of creating a new event. The user will have to remember not to change the eventId once set, otherwise a new event would be created on Google Calendar.
+* The user will have to specify the Calendar Identifier where the event will be created. The organizer must have access to that calendar.
```

Comment:
Can we skip this and instead add a step where an authorized administrator of the GitHub and of the calendar would connect the two using the GitHub Action?

Reply:
I think we could check if the `calendarid` is set or not. If it's not set we would query the Calendar API for all the organizer's calendars and show a failure message to the user such as:

```
We could not create event, calendarId not set.

Find the calendarId in the table below and set in y2c.yml

 Name        | ID
-------------+------------
 Kubernetes  | calendar-1
 Prometheus  | calendar-2
```

Reply:
I'm envisioning a one-to-one mapping of org to Google Calendar, so I don't think this functionality makes sense. Especially at launch.

Reply:
@dankohn don't quite follow. The `calendarId` is required https://developers.google.com/calendar/v3/reference/events/insert. So if you want to create an event in a specific calendar the `calendarId` must be specified. If it's not important to add events to a specific calendar, we could automatically create one `Y2C` calendar and use that instead.

Reply:
OK, I'm convinced we need calendarID.